Source/WebKit/GPUProcess/webrtc/RemoteSampleBufferDisplayLayer.mm

```diff
@@ -110,12 +110,26 @@
     protectedSampleBufferDisplayLayer()->updateDisplayMode(hideDisplayLayer, hideRootLayer);
 }
 
-void RemoteSampleBufferDisplayLayer::updateBoundsAndPosition(CGRect bounds, std::optional<WTF::MachSendRight>&& fence)
+void RemoteSampleBufferDisplayLayer::updateBoundsAndPosition(CGRect bounds, std::optional<WTF::MachSendRightAnnotated>&& fence)
 {
-    if (fence && fence->sendRight())
-        m_layerHostingContext->setFencePort(fence->sendRight());
+#if USE(EXTENSIONKIT)
+    RetainPtr<BELayerHierarchyHostingTransactionCoordinator> hostingUpdateCoordinator;
+    if (fence && fence->sendRight.sendRight()) {
+#if ENABLE(MACH_PORT_LAYER_HOSTING)
+        hostingUpdateCoordinator = LayerHostingContext::createHostingUpdateCoordinator(WTFMove(*fence));
+#else
+        hostingUpdateCoordinator = LayerHostingContext::createHostingUpdateCoordinator(fence->sendRight.sendRight());
+#endif // ENABLE(MACH_PORT_LAYER_HOSTING)
+        [hostingUpdateCoordinator addLayerHierarchy:m_layerHostingContext->hostable().get()];
+    }
+    protectedSampleBufferDisplayLayer()->updateBoundsAndPosition(bounds, { });
+    [hostingUpdateCoordinator commit];
```

Comment:
`hostingUpdateCoordinator` may be nil. Is it okay? Sending message to nil is acceptable but smelling bad behavior.

Reply:
That's a good point! I added checks and an error message.

Thanks for reviewing!